### DISCUSSION TITLE
LAB 02.03 Task 3: Fill in missing data

### DISCUSSION THREAD

**Discussion by sarasti2** [rramosp/ai4eng.v1#272] [02 - PYTHON]
Tengo la siguiente función para resolver el problema:
```
def fillna(df):
    # make sure to make a copy in case you modify the original df
    df1 = df.copy()
    mean = df[df.price.notna()]['price'].mean()
    std = df[df.price.notna()]['price'].std()
    df1['price'].fillna(np.random.normal(mean,std), inplace = True)
    return df1
    
```
al verificar manualmente funciona perfecto, sin embargo, la plataforma me devuelve una nota de 0, porque la media que ella calcula es diferente a la de mis datos, pero no encuentro donde está el error, me gustaría saber si alguien tiene alguna idea de por qué puede estar pasando

**Comment by sarasti2** [comment]
Error encontrado, al hacerlo así:
` df1['price'].fillna(np.random.normal(mean,std), inplace = True)`
todos se remplazan por el mismo número

**Comment by rramosp** [comment]
exacto! perfecto alejandro

On Sat, Sep 9, 2023 at 1:38 AM Alejandro Sarasti Sierra <
***@***.***> wrote:

> Error encontrado, al hacerlo así:
> df1['price'].fillna(np.random.normal(mean,std), inplace = True)
> todos se remplazan por el mismo número
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rramosp/ai4eng.v1/discussions/272#discussioncomment-6953995>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADJRZIMOKUQ6ZGFFUJOHVFLXZQFGJANCNFSM6AAAAAA4RHQ2C4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

-- 
*Aviso legal:* El contenido de este mensaje y los archivos adjuntos son 
confidenciales y de uso exclusivo de la Universidad Nacional de Colombia. 
Se encuentran dirigidos sólo para el uso del destinatario al cual van 
enviados. La reproducción, lectura y/o copia se encuentran prohibidas a 
cualquier persona diferente a este y puede ser ilegal. Si usted lo ha 
recibido por error, infórmenos y elimínelo de su correo. Los Datos 
Personales serán tratados conforme a la Ley 1581 de 2012 y a nuestra 
Política de Datos Personales que podrá consultar en la página web 
www.unal.edu.co <http://www.unal.edu.co/>.* *Las opiniones, informaciones, 
conclusiones y cualquier otro tipo de dato contenido en este correo 
electrónico, no relacionados con la actividad de la Universidad Nacional de 
Colombia, se entenderá como personales y de ninguna manera son avaladas por 
la Universidad.
